woocommerce/woocommerce-android
# [Payment Method Improvements] Create the "Record transaction details in order note" section

Adresses: #11525

Also contains changes from this PR which have already been reviewed:

https://github.com/woocommerce/woocommerce-android/pull/11570

Follow up items from that PR are in this ticket at TODOs:
https://github.com/woocommerce/woocommerce-android/issues/11583


### Description

Add the toggle for "Record transaction details in order note" in the Change Due Calculator screen.

Also addresses some feedback from the previous PR that @JorgeMucientes provided me with:

* Fix toolbars: https://github.com/woocommerce/woocommerce-android/pull/11555#discussion_r1611985517
* Fix text field design: https://github.com/woocommerce/woocommerce-android/pull/11555#discussion_r1612008744
* Update navigation pattern to be in line with the rest of the app: https://github.com/woocommerce/woocommerce-android/pull/11555#discussion_r1612016240
* Change due field should not appear to be editable: "My understanding is that the Change due field is non editable right?"


### Testing instructions


0. Enable OTHER_PAYMENT_METHODS
1. Go to an order that needs to be paid
2. Select "Cash"
3. Make sure toolbar back button works

Expected: You should see the Change Due screen


No added functionality, that will be added [here](https://github.com/woocommerce/woocommerce-android/issues/10941). UI should look similar to iOS:

![IMG_0447](https://github.com/woocommerce/woocommerce-android/assets/947561/22ce6d80-78d6-4e4d-8085-bc0030a6dce6)

These issues should be fixed:

- [ ]  Only one toolbar visible: https://github.com/woocommerce/woocommerce-android/pull/11555#discussion_r1611985517
- [ ]  Fix text field design: https://github.com/woocommerce/woocommerce-android/pull/11555#discussion_r1612008744
- [ ]  Navigation pattern should be in line with the rest of the app: https://github.com/woocommerce/woocommerce-android/pull/11555#discussion_r1612016240
- [ ]  Change due field should not appear to be editable
